### PR TITLE
Deprecated getAsReadOnly() methods in configuration classes

### PR DESCRIPTION
**Pull request by Donnerbart**
* deprecated all `getAsReadOnly()` and `asReadOnly()` methods
* made new `getAsReadOnly()` methods from 3.8 package private
* moved package private methods down to the read-only class definition
* moved `DurableExecutorConfigTest` to the config package

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/config/CardinalityEstimatorConfig.java [hunk 1]
```diff
@@ -153,8 +146,14 @@ public String toString() {
                 + asyncBackupCount + ", readOnly=" + readOnly + '}';
     }
 
-    private static class CardinalityEstimatorConfigReadOnly
-            extends CardinalityEstimatorConfig {
+    CardinalityEstimatorConfigReadOnly getAsReadOnly() {
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/config/CardinalityEstimatorConfig.java]: Can't this be made private?

File: hazelcast/src/main/java/com/hazelcast/config/DurableExecutorConfig.java [hunk 2]
```diff
@@ -171,9 +164,16 @@ public String toString() {
                 + '}';
     }
 
+    DurableExecutorConfigReadOnly getAsReadOnly() {
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/config/DurableExecutorConfig.java]: private?

File: hazelcast/src/main/java/com/hazelcast/config/NearCachePreloaderConfig.java [hunk 3]
```diff
@@ -162,6 +155,13 @@ public String toString() {
                 + '}';
     }
 
+    NearCachePreloaderConfig getAsReadOnly() {
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/config/NearCachePreloaderConfig.java]: private?

**Comment by pveentjer** [comment]
:+1: good work

File: hazelcast/src/main/java/com/hazelcast/config/CardinalityEstimatorConfig.java (hunk 1 above)
Comment (Donnerbart) [inline hazelcast/src/main/java/com/hazelcast/config/CardinalityEstimatorConfig.java]: Those methods are used in tests for `UnsupportedOperationException` in the read-only configs, so none of them can be private.

**Pull request merged**